src/app/parcels/ActionBar/DuplicateDownloadWarning.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+
+import { getParcelsWithEvent } from "@/app/parcels/parcelsTable/fetchParcelTableData";
+import Alert from "@mui/material/Alert/Alert";
+import styled from "styled-components";
+import { StatusType } from "@/app/parcels/ActionBar/Statuses";
+import { ParcelsTableRow } from "@/app/parcels/parcelsTable/types";
+
+export interface DuplicateDownloadWarningProps {
+    parcels: ParcelsTableRow[];
+    targetEventName: StatusType;
+}
+
+const StyledAlert = styled(Alert)`
+    border-radius: 4px;
+`;
+
+const DuplicateDownloadWarning: React.FC<DuplicateDownloadWarningProps> = async (props) => {
+    const parcelIds = props.parcels.map((parcel) => parcel.parcelId);
+    const { parcels, error } = await getParcelsWithEvent(props.targetEventName, parcelIds);
+    if (error) {
+        return;
+    }
+
+    const postcodes = parcels.map((parcel) => parcel.client_address_postcode);
+    const uniquePostcodes = Array.from(new Set(postcodes));
+
+    return (
+        uniquePostcodes.length > 0 && (
+            <StyledAlert severity="warning">
+                The following postcodes have already been downloaded: {uniquePostcodes.join(", ")}.
+                Are you sure you want to print again?
+            </StyledAlert>
+        )
+    );
```

Comment:
Wording like that to me implies a confirmation window
Does the warning appear after you press print or just when you open the modal? I think it's better if it appears when you press the button, and then the button text turns into "Confirm Download PDF"
If it's the other way this might need some rewording but I think I prefer the confirmation way (I'm actually not sure - is it meant to show a warning like that or have another window pop up? I guess both patterns get used in the wild)

Let me know your thoughts maybe I'm misinterpreting something

Reply:
Currently it just appears when you open the modal. Adjusting the wording would be much simpler than changing the button's behaviour but I'm happy to do whichever you think is better. I guess you could also argue that this modal is a kind of confirmation anyway but I would tend to agree with you that it implies a separate confirmation step.

Reply:
Personally I feel showing the text before clicking the button is what Dan wanted, as this is the behaviour that exists already in another action

Reply:
Ok, let's keep it as is :)